modules/indexer/issues/indexer.go

```diff
@@ -26,9 +25,24 @@ import (
 	"code.gitea.io/gitea/modules/util"
 )
 
+// IndexerMetadata is used to send data to the queue, so it contains only the ids.
+// It may look weired, because it has to be compatible with the old queue data format.
+// If the IsDelete flag is true, the IDs specify the issues to delete from the index without querying the database.
+// If the IsDelete flag is false, the ID specify the issue to index, so Indexer will query the database to get the issue data.
+// It should be noted that if the id is not existing in the database, it's index will be deleted too even if IsDelete is false.
+// Valid values:
+//   - IsDelete = true, IDs = [1, 2, 3], and ID will be ignored
+//   - IsDelete = false, ID = 1, and IDs will be ignored
+type IndexerMetadata struct {
+	ID int64 `json:"id"`
+
+	IsDelete bool    `json:"is_delete"`
+	IDs      []int64 `json:"ids"`
+}
```

Comment:
I think we can introduce a "Version" field here.

If Version=0, then use old rules.

If Version=1, then use new rules. New code will always emit Version=1 items.

Reply:
Since the new struct is compatible with the old one, we could do that if we have to break it one day.

Although it looks weird, I think it's good enough.